### PR TITLE
[1.0] Resolve tpm2-tss CVE-2023-22745

### PR DESCRIPTION
**Pull request by henryli001**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
- Add patch to resolve CVE-2023-22745

###### Change Log  

- Resolve CVE-2023-22745

###### Does this affect the toolchain?  

NO

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2023-22745

###### Test Methodology

- Local Build

### CODE REVIEW

File: SPECS/tpm2-tss/tpm2-tss.signatures.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
  "Signatures": {
-  "tpm2-tss-2.4.6.tar.gz": "20e6da532a7ef90c8e50cca51f276053ec505eee0167c18e2b07c1e747118b58"
```
Comment (dmcilvaney) [inline SPECS/tpm2-tss/tpm2-tss.signatures.json:3]: This is a significant version bump for 1.0, is there any  hope of backporting the patch for this?

Comment (henryli001) [reply SPECS/tpm2-tss/tpm2-tss.signatures.json]: Good point. Looked into it and seems like the patch can be applied to the current source code. Have re-implemented the patch.

File: SPECS/nodejs/nodejs.signatures.json [hunk 2]
```diff
@@ -1,6 +1,5 @@
 {
-  "Signatures": {
-    "clean-source-tarball.sh": "ab579872ec2f4e85a00fc1651b40a26876012256c722c3701ef3cdcb378c93d5",
-    "node-v14.21.1.tar.xz": "51bce5965dfb20764ed6fb7855eb04f1eae41a02fb8422f9dbe2c268a8b2e61e"
-  }
+ "Signatures": {
+  "node-v14.21.1.tar.xz": "51bce5965dfb20764ed6fb7855eb04f1eae41a02fb8422f9dbe2c268a8b2e61e"
+ }
```
Comment (dmcilvaney) [inline SPECS/nodejs/nodejs.signatures.json:4]:
```suggestion
 "Signatures": {
  "clean-source-tarball.sh": "ab579872ec2f4e85a00fc1651b40a26876012256c722c3701ef3cdcb378c93d5",
  "node-v14.21.1.tar.xz": "51bce5965dfb20764ed6fb7855eb04f1eae41a02fb8422f9dbe2c268a8b2e61e"
 }
```

Comment (dmcilvaney) [reply SPECS/nodejs/nodejs.signatures.json:4]: I'd not include this here, this will probably  get cleaned up the next time someone modifies nodejs

**Review by jslobodzian — APPROVED** [review]

**Pull request merged**